### PR TITLE
 thunderbird, thunderbird-bin: 68.10.0 -> 78.0 [High security fixes]

### PR DESCRIPTION
**Pull request by taku0**
###### Motivation for this change

- High security fixes
- Various updates

https://www.mozilla.org/en-US/security/advisories/mfsa2020-29/
https://www.thunderbird.net/en-US/thunderbird/78.0/releasenotes/

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by xfix** [comment]
Can we avoid backporting this at least until PGP support gets implemented? As release notes say "At this time, users of the Enigmail Add-on should not update to Thunderbird 78", and I personally depend on Enigmail.

**Comment by vcunat** [comment]
I agree.  AFAIK it doesn't auto-upgrade yet (e.g. the Windows version, I mean).  I expect 68 will be supported and default at least until this stuff is resolved upstream.

**Comment by taku0** [comment]
Now expressions for both 68 (default) and 78 (with suffix) are available.

**Review by lovesegfault — APPROVED** [review]

**Comment by mweinelt** [comment]
LGTM. Did a quick run-test on `thunderbird-78`.

Result of `nixpkgs-review pr 93227` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
<br>- thunderbird
<br>- thunderbird-78
<br>- thunderbird-bin-78
</details>

**Review by mweinelt — APPROVED** [review]

File: pkgs/applications/networking/mailreaders/thunderbird/default.nix [hunk 1]
```diff
@@ -223,12 +221,9 @@ stdenv.mkDerivation rec {
     "--enable-default-toolkit=${toolkitValue}"
     "--enable-js-shell"
     "--enable-necko-wifi"
-    "--enable-startup-notification"
     "--enable-system-ffi"
     "--enable-system-pixman"
-    "--enable-system-sqlite"
 
-    "--disable-gconf"
```
Comment (vcunat) [inline pkgs/applications/networking/mailreaders/thunderbird/default.nix:231]: Can we mention somewhere why these options were removed? (in the commit message perhaps)

Comment (taku0) [reply pkgs/applications/networking/mailreaders/thunderbird/default.nix:231]: Updated the commit message.

**Comment by taku0** [comment]
Is Thunderbird 78.1 coming?

https://www.mozilla.org/en-US/security/advisories/mfsa2020-33/

**Comment by vcunat** [comment]
I saw it yesterday, but there are still no release notes or tarballs even now.

**Comment by xfix** [comment]
https://www.thunderbird.net/en-US/thunderbird/78.1.0/releasenotes/

**Comment by vcunat** [comment]
Still not suitable for default.
> At this time, users of the Enigmail Add-on should not update to Thunderbird 78.
>
> OpenPGP in Thunderbird 78.1.0 has reached feature complete state, but it's still disabled by default, to allow more time for testing, correctness, and localization. See the wiki for how to enable and help with testing.

**Review by vcunat — APPROVED** [review]
Both seem to work for me, but so far I haven't been brave enough to start that enigmail migration wizard.

File: pkgs/applications/networking/mailreaders/thunderbird/default.nix (hunk 1 above)
Comment (vcunat) [reply pkgs/applications/networking/mailreaders/thunderbird/default.nix:231]: `-icu` was there twice.  For the rest I can't see them referenced in the output, so I assume the flags being unsupported means that thunderbird now always bundles those dependencies :-/  I expect we should remove them from build inputs (some time).

**Comment by vcunat** [comment]
Well, one of the extensions I've been using isn't compatible, but such things aren't really a surprise.

**Comment by mweinelt** [comment]
> I haven't been brave enough to start that enigmail migration wizard.

I have and it didn't work. Exported my private key and it's a 2048 bit RSA key, 260kB in size, I get an error when trying to import it manually. 

Specifically I'm running into https://bugzilla.mozilla.org/show_bug.cgi?id=1654894.

**Comment by ShamrockLee** [comment]
FYI
thunderbird, thunderbird-bin: 68.10.0 -> 68.11.0, 78.1.0 -> 78.1.1 [High security fixes] #94863

**Pull request merged**